Lib/posixpath.py

```diff
@@ -202,26 +199,14 @@ def ismount(path):
         if stat.S_ISLNK(s1.st_mode):
             return False
 
-    path = os.fspath(path)
-    if isinstance(path, bytes):
-        parent = join(path, b'..')
-    else:
-        parent = join(path, '..')
-    parent = realpath(parent)
+    parent = dirname(abspath(path))
```

Comment:
I think replacing `realpath(join(path, '..'))` with `dirname(abspath(path))` fine, because we've already established that _path_ is not a symlink, and therefore its lexical parent is guaranteed to be its canonical parent. But I'm calling it out as I'd like another reviewer to check my logic here!

Reply:
In fact, perhaps you could split this change out into a separate PR, if that's not too much trouble?

It's unique in that you've actually reduced the number of system calls made (which is awesome), whereas the other changes in this PR are micro-optimizations of the Python code.

Reply:
Nope that's fine.